src/phase0/state_transition/slot_processing.rs

```diff
@@ -67,11 +75,137 @@ pub fn process_slot<
 
     let previous_state_root = state.hash_tree_root(merkleization_context)?;
 
-    state.state_roots[(state.slot % context.slots_per_historical_root as u64) as usize] = previous_state_root;
+    state.state_roots[(state.slot % context.slots_per_historical_root as u64) as usize] =
+        previous_state_root;
     if state.latest_block_header.state_root.as_ref() == Bytes32::default().as_ref() {
         state.latest_block_header.state_root = previous_state_root;
     }
-    let previous_block_root = state.latest_block_header.hash_tree_root(merkleization_context)?;
-    state.block_roots[(state.slot % context.slots_per_historical_root as u64) as usize] = previous_block_root;
+    let previous_block_root = state
+        .latest_block_header
+        .hash_tree_root(merkleization_context)?;
+    state.block_roots[(state.slot % context.slots_per_historical_root as u64) as usize] =
+        previous_block_root;
     Ok(())
-}
\ No newline at end of file
+}
+
+pub fn state_transition<
```

Comment:
another stylistic thing but i'd personally put a function like this into the `phase0::state_transition` top-level module, not these "sub-modules" for the individual sub-transitions

Reply:
fixed